demos/cms/cms_comp.c

```diff
@@ -48,11 +48,11 @@ int main(int argc, char **argv)
     if (!SMIME_write_CMS(out, cms, in, flags))
         goto err;
 
-    ret = 0;
+    ret = EXIT_SUCCESS;
 
  err:
 
-    if (ret) {
+    if (ret == EXIT_FAILURE) {
```

Comment:
(very minor) The equivalent to the previous condition (`if (ret)`, i.e. `if (ret != 0)`) is slightly better as `if (ret != EXIT_SUCCESS)`, both here and in other places.

Reply:
ret is either EXIT_FAILURE or EXIT_SUCCESS, so checking for either
if it doesnt pass OR if it failed are equivalent.

Reply:
I also think testing this way is the safer option.  Any non-success value triggers something.